tests/postgres_tests/test_indexes.py

```diff
@@ -665,6 +665,30 @@ def create_sql(self, model, schema_editor, using="gin", **kwargs):
                 str(index.create_sql(CharFieldModel, editor)),
             )
 
+    def test_custom_sql(self):
+        class CustomSQLIndex(PostgresIndex):
+            sql_create_index = "SELECT 1"
+            sql_delete_index = "SELECT 2"
+
+            def create_sql(self, model, schema_editor, using="", **kwargs):
+                kwargs.setdefault("sql", self.sql_create_index)
+                return super().create_sql(model, schema_editor, using, **kwargs)
+
+            def remove_sql(self, model, schema_editor, **kwargs):
+                kwargs.setdefault("sql", self.sql_delete_index)
+                return super().remove_sql(model, schema_editor, **kwargs)
+
+        index = CustomSQLIndex(fields=["field"], name="custom_sql_idx")
+
+        operations = [
+            (index.create_sql, CustomSQLIndex.sql_create_index),
+            (index.remove_sql, CustomSQLIndex.sql_delete_index),
+        ]
+        for operation, expected in operations:
+            with self.subTest(operation=str(operation)):
```

Comment:
I'd use
```suggestion
            with self.subTest(operation=operation.__class__.__name__).:
```

Reply:
Good suggestion, though perhaps you meant `operation.__name__`? Otherwise the test shows `operation='method'` on error:

```
======================================================================
FAIL: test_custom_sql (postgres_tests.test_indexes.SchemaTests.test_custom_sql) [...] (operation='method')
```

Using `operation.__name__` it prints:
```
======================================================================
FAIL: test_custom_sql (postgres_tests.test_indexes.SchemaTests.test_custom_sql) [...] (operation='create_sql')
[...]
======================================================================
FAIL: test_custom_sql (postgres_tests.test_indexes.SchemaTests.test_custom_sql) [...] (operation='remove_sql')
```

Reply:
> Good suggestion, though perhaps you meant `operation.__name__`? Otherwise the test shows `operation='method'` on error:

Right, sorry :+1: